### PR TITLE
Explain focus problems and description of algorithms for LCT module

### PR DESCRIPTION
**Pull request by Gabriel-Xu**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
this isn't meant to be complete right

**Comment by Gabriel-Xu** [comment]
> this isn't meant to be complete right

I'm not sure exactly what you are looking for since the instructions said to explain the focus problems and describe the algorithms. There was one focus problem that I thought didn't require much explanation if you understand the material provided in the resources. I have seen many currently complete modules leave the explanation of algorithms mostly to the attached resources, so I could explain the part about sum/max/min over paths more, but otherwise I'm not sure what part I should add to.

**Comment by ryanchou-dev** [comment]
Could you explain the sample problems on the module? (like LCT - Subtrees)

Let me know if you have questions!

**Comment by Gabriel-Xu** [comment]
> Could you explain the sample problems on the module? (like LCT - Subtrees)
> 
> Let me know if you have questions!

Does explaining the problem mean adding a code snippet in addition to a text explanation? Also, does sample problems refer specifically to the Vertex Add Path Sum problem and Vertex Add Subtree Sum problem?

**Comment by ryanchou-dev** [comment]
Yep.

**Comment by ryanchou-dev** [comment]
any updates?

**Comment by Gabriel-Xu** [comment]
> any updates?

I am working on it

**Comment by Gabriel-Xu** [comment]
Could someone review my new commits?

File: content/6_Advanced/LCT.mdx [hunk 1]
```diff
@@ -399,6 +657,224 @@ int main() {
...
+We can also maintain information about subtrees by keeping track of values for the virtual subtrees of a node. When querying for information such as the subtree sum, we call $\textup{access}()$ on the node so that all of its children in the represented tree are part of virtual subtrees and then retrieve the desired value.
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
+
+// BeginCodeSnip{Benq Link Cut Tree}
+/**
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:671]: if this impl is linked in the resources, remove this doc comment

File: content/6_Advanced/LCT.mdx [hunk 2]
```diff
@@ -399,6 +657,224 @@ int main() {
 	/>
 </Resources>
 
+### Implementation
+
+We can also maintain information about subtrees by keeping track of values for the virtual subtrees of a node. When querying for information such as the subtree sum, we call $\textup{access}()$ on the node so that all of its children in the represented tree are part of virtual subtrees and then retrieve the desired value.
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:662]: uh i think we use texttt instead of textup

File: content/6_Advanced/LCT.mdx [hunk 3]
```diff
@@ -357,14 +388,239 @@ int main() {
...
+	friend void cut(sn x, sn y) {  // if x, y adj in tree
+		x->makeRoot();
+		y->access();
+		assert(y->c[0] == x && !x->c[0] && !x->c[1]);
+		cut(y);
+	}
+};
+// EndCodeSnip
+
+const int maxn = 2e5 + 5;
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:583]: rename to MAX_N
also is the +5 necessary

File: content/6_Advanced/LCT.mdx [hunk 4]
```diff
@@ -357,14 +388,239 @@ int main() {
...
+Link Cut Tree can also calculate the aggregate (max, min, sum, etc.) of the weights of the edges or nodes on a path. To do this, we can define a function that will return an aggregate for the path from the root of the tree to the provided vertex. We can augment the auxiliary splay trees with the value(s) we want to keep track of. The aggregate for the path from the root of a tree to a vertex can be found by retrieving the value(s) from the splay tree created after accessing the vertex.
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
+
+// BeginCodeSnip{Benq Link Cut Tree}
+/**
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:414]: same thing w the comment here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LCT.mdx (hunk 1 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Fixed

File: content/6_Advanced/LCT.mdx (hunk 2 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Fixed

File: content/6_Advanced/LCT.mdx (hunk 3 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Fixed, the +5 is not necessary

File: content/6_Advanced/LCT.mdx (hunk 4 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Fixed

File: content/6_Advanced/LCT.mdx [hunk 5]
```diff
@@ -399,6 +637,204 @@ int main() {
...
+		int a;
+		cin >> a;
+		LCT[i] = new snode(a);
+	}
+	for (int i = 0; i < n - 1; i++) {
+		int u, v;
+		cin >> u >> v;
+		link(LCT[u], LCT[v], 1);
+	}
+	while (q--) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:813]: can you use a for loop instead (and the other sol too)

Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Fixed

File: content/6_Advanced/LCT.mdx [hunk 6]
```diff
@@ -41,6 +44,20 @@ frequency: 1
...
+can perform the following operations with an amortized upper bound time
+complexity of $\mathcal{O}(\log N)$:
+
+- Linking a tree with a node by making the root of the tree a child of any node
+  of another tree
+- Deleting the edge between a node and its parent, detaching the node's subtree
+  to make a new tree
+- Find the root of the tree that a node belongs to
+
+These operations all use the $\text{access}(v)$ subroutine, which creates a preferred path from the root of the represented tree to vertex $v$, making a corresponding auxiliary splay tree with $v$ as the root.
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:57]:
```suggestion
These operations all use the $\texttt{access}(v)$ subroutine,
which creates a preferred path from the root of the represented tree to vertex $v$,
making a corresponding auxiliary splay tree with $v$ as the root.
```

File: content/6_Advanced/LCT.mdx [hunk 7]
```diff
@@ -357,14 +388,219 @@ int main() {
...
+			cin >> u >> v >> w >> x;
+			cut(LCT[u], LCT[v]);
+			link(LCT[w], LCT[x], 1);
+		} else if (t == 1) {
+			int p, x;
+			cin >> p >> x;
+			LCT[p]->access();
+			LCT[p]->val += x;
+			LCT[p]->calc();
+		} else {
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:593]:
```suggestion
		} else if (t == 2) {
```

File: content/6_Advanced/LCT.mdx [hunk 8]
```diff
@@ -399,6 +637,204 @@ int main() {
...
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:827]:
```suggestion
		} else if (t == 2) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LCT.mdx [hunk 9]
```diff
@@ -386,6 +624,8 @@ int main() {
 
 ## Link Cut Tree - Subtrees
 
+### Sample Problem
+
 <Problems problems="subSam" />
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:629]: make this a focus problem?

**Comment by SansPapyrus683** [comment]
i made some changes myself, please be sure to pull!

File: content/6_Advanced/LCT.mdx (hunk 9 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Done

**Comment by Gabriel-Xu** [comment]
> i made some changes myself, please be sure to pull!

Okay, your changes are all added.

File: content/6_Advanced/LCT.mdx [hunk 10]
```diff
@@ -355,16 +388,226 @@ int main() {
...
 ```
 
-## Link Cut Tree - Connectivity
+## Finding Connectivity
+
+The link cut tree can be used to solve problems that deal with queries updating trees and querying for connectivity.
+
+### Problems
 
 <Problems problems="con" />
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:397]: @ryanchou-dev this part is just two problems...i'm thinking smth like either delete both or keep just one and explain it

File: content/6_Advanced/LCT.mdx [hunk 11]
```diff
@@ -215,7 +233,22 @@ int main() {
 }
 ```
 
-### With Euler-Tour Tree
+### With Euler Tour Tree
+
+An alternative solution is to use the Euler Tour Tree (ETT) technique.
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:238]: i believe the usaco guide has a module on the euler tour, can you link that?

File: content/6_Advanced/LCT.mdx [hunk 12]
```diff
@@ -355,16 +388,226 @@ int main() {
...
+
+### Explanation
+
+Link Cut Tree can also calculate the aggregate (max, min, sum, etc.) of the weights of the edges or nodes on a path. To do this, we can define a function that will return an aggregate for the path from the root of the tree to the provided vertex. We can augment the auxiliary splay trees with the value(s) we want to keep track of. The aggregate for the path from the root of a tree to a vertex can be found by retrieving the value(s) from the splay tree created after accessing the vertex.
 
 ### Implementation
 
+**Time Complexity:** $\mathcal{O}(?)$
+
+```cpp
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:411]: around all cpp code, please wrap it with `<LanguageSection>` and `<CPPSection>`
check the other modules for how they do stuff

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LCT.mdx (hunk 12 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx:427]: Done

File: content/6_Advanced/LCT.mdx (hunk 11 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx:244]: Should I add it as a resource or just a normal hyperlink?

Comment (SansPapyrus683) [reply content/6_Advanced/LCT.mdx:241]: just a normal hyperlink
but i'm dumb so can you check if this euler tour is the same as the one covered in the guide? if it isn't then add an explanation on how this differs from the other euler tour

Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx]: Done

**Comment by SansPapyrus683** [comment]
alright can you test the updated solutions and show if they still work

**Comment by Gabriel-Xu** [comment]
> alright can you test the updated solutions and show if they still work

Yes, the solutions work

**Comment by SansPapyrus683** [comment]
may i see proof?

**Comment by Gabriel-Xu** [comment]
> may i see proof?

https://judge.yosupo.jp/submission/206454
https://judge.yosupo.jp/submission/206455
AC

**Review by SansPapyrus683 — APPROVED** [review]
i'm allowing the benq ahh data structure only because i haven't made a sansified version yet

File: content/6_Advanced/LCT.mdx (hunk 10 above)
Comment (ryanchou-dev) [reply content/6_Advanced/LCT.mdx:408]: @Gabriel-Xu are you cool with explaining one of these problems briefly?

i think explanation would be better than just deleting the section

**Review by ryanchou-dev — COMMENTED** [review]
otherwise all good!

File: content/6_Advanced/LCT.mdx (hunk 10 above)
Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx:408]: > @Gabriel-Xu are you cool with explaining one of these problems briefly?
> 
> i think explanation would be better than just deleting the section

Do you want it to be another focus problem?

Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx:408]: Also does this mean I should include a code solution or just a text explanation?

Comment (Gabriel-Xu) [reply content/6_Advanced/LCT.mdx:408]: For now, I added a brief explanation.

**Comment by Gabriel-Xu** [comment]
Can someone please review this?

**Comment by ryanchou-dev** [comment]
> Can someone please review this?

Looks good. I think the intention was for the problem to be a focus problem with a brief explanation below. If you already have code, you could add that as well.

You can delete the "Silence's Sounds" problem. Everything else looks great, so I'll merge after that 🙏

**Comment by Gabriel-Xu** [comment]
> > Can someone please review this?
> 
> Looks good. I think the intention was for the problem to be a focus problem with a brief explanation below. If you already have code, you could add that as well.
> 
> You can delete the "Silence's Sounds" problem. Everything else looks great, so I'll merge after that 🙏

Okay, I made it a focus problem and added the code, and deleted the "Silence's Sounds" problem.

**Comment by Gabriel-Xu** [comment]
@ryanchou-dev  is it good?

**Comment by Gabriel-Xu** [comment]
Please take a look soon

**Review by ryanchou-dev — APPROVED** [review]
perfect! thanks for all of your contributions :)

**Comment by Gabriel-Xu** [comment]
Thanks!

**Comment by SansPapyrus683** [comment]
uh so we can pay through paypal or amazon gift card
send email w/ the relevant info to [handle] AT gmail.com
[handle] = kevinsheng2005

**Pull request merged**